### ISSUE TITLE
[Possible Feature] Multiplying Volume to the Pumps and Dumps

### ISSUE THREAD

**Issue by patbaumgartner** [brianleect/binance-pump-alerts#29]
Usually when bigger moves on the charts are starting, there is already quite a lot of volume. Maybe it's a good idea to add volume to the game. Maybe using it as a multiplier ....

Hope we can discuss this topic here.

**Comment by brianleect** [comment]
Previously mentioned threads regarding volume implementation, https://github.com/brianleect/binance-pump-alerts/issues/24 https://github.com/brianleect/binance-pump-alerts/issues/10 https://github.com/brianleect/binance-pump-alerts/issues/12

Volume is definitely a good indicator of big moves in crypto. However, there are some limitations regarding the current way the program is structured.

1. Weight Limits https://github.com/brianleect/binance-pump-alerts/issues/12 https://github.com/brianleect/binance-pump-alerts/issues/10
2. 24h rolling volume does not show clear representation of volume https://github.com/brianleect/binance-pump-alerts/issues/10

I'm currently unaware of any REST endpoints that can provide the volume information required.

Potential Solution
1. Usage of websockets instead of REST API
1a. With websocket which send trades real-time, we theoretically would be able to get any changes in volume. 

Reference: https://github.com/binance/binance-spot-api-docs/blob/master/web-socket-streams.md#aggregate-trade-streams

However, I'm not familiar with the usage of binance websocket. To my knowledge, there would be a need to have the websocket setup to connect to all pairs on Binance and process the information simultaneously to have a similar effect to what we have with a REST API implementation.

**Comment by patbaumgartner** [comment]
I see your point and totally agree with the limitations. 

When sticking the rest API (which currently uses no API Keys), there may be needs a two-step approach. But even when you filter most of the pairs, you still have plenty of calls to retrieve the volume.

**Comment by patbaumgartner** [comment]
Closed accidentially. Keep it open.

**Comment by brianleect** [comment]
> I see your point and totally agree with the limitations. Thanks for your detailed explanation.
> 
> When sticking the rest API (which currently uses no API Keys), there may be needs a two-step approach. But even when you filter most of the pairs, you still have plenty of calls to retrieve the volume.
> 
> Needs more thinking, I guess.

Yep at 1200 requests / minute you have quite a decent amount of room to play with. Assuming we look at purely USDT pairs, we have around 260 of them. Assuming each "volume call" has a weight of 1, you can call each pair around 4-5 times per minute. So it would theoretically work with '15s' interval.

However, if you were looking to monitor all pairs on the exchange it would be close to >2k pairs which would limit it even further.

We would unlikely be able to maintain the extraction frequency of '1s' unless the number of pairs being monitored is heavily cut down.
